### ISSUE TITLE
Failing page build for OpenRefine lesson

### ISSUE THREAD

**Issue by mdlincoln** [programminghistorian/jekyll#565]
Well I've looked up and down but for the life of me I can't understand what is causing GH pages to fail the build when the latest lesson "Fetching and Parsing Data from the Web with OpenRefine" is added in the repository. The site builds perfectly well locally, on Travis, and on Netlify. 

I am going to keep looking through troubleshooting docs and see if I can pin this down. Any further related stuff let's log in this issue.

**Comment by jerielizabeth** [comment]
Sounds good. I am also finding that it builds just fine locally, and on Travis and Netlify. 

When I removed the Open Refine files, it still failed to build on the gh-pages branch. Is there any chance that there was an update on the Github side of things that is complicating our troubleshooting?

**Comment by jerielizabeth** [comment]
Question for clarification: Is it correct that with the new theme, we are no longer hand creating the lesson index page, but that it is generated from the files in the lessons folder?

**Comment by jerielizabeth** [comment]
Huzzah! We have a new error!! Failure to build on netlify.

[update]: it built again as of 10:40. Gremlins I say!

**Comment by jerielizabeth** [comment]
I am seeing one piece of strangeness on the local build in the dates. It is showing the correct published date (PUBLISHED 2017-08-12) but an earlier modified date (MODIFIED 2017-07-14).

**Comment by jerielizabeth** [comment]
Should we go back to last commit that built? (https://github.com/programminghistorian/jekyll/commit/b8f1acb4898372130e489ded3485b9b2a47d3014)

**Comment by mdlincoln** [comment]
@jerielizabeth let me fiddle a bit more with the lesson file first

**Comment by mdlincoln** [comment]
I've logged an issue with GitHub support.

**Comment by mdlincoln** [comment]
Received this morning:

---

Hey Matt,

Thanks for reaching out, and sorry for the delayed response.

We've asked our engineers to help dig into the cause of this. I can't promise an ETA, but we'll let you know as soon as we have any updates to share.

Cheers,
Antonio
GitHub Support

**Comment by mdlincoln** [comment]
Received today - trying another build in a moment:

---

Hi Matthew,

We just upgraded to Jekyll 3.5.2 yesterday which may resolve this issue. Please try another build of your site and let us know if you continue to receive errors.

Thanks,

Steve
@slgraff
GitHub Support

**Comment by mdlincoln** [comment]
... Still broken.

**Comment by mdlincoln** [comment]
This is the song that never ends...

---

Shawna,
 
Thanks for the suggestion! As you’ll note in our config file (https://github.com/programminghistorian/jekyll/blob/gh-pages/_config.yml#L14-L15), we don’t use the jekyll-seo-tag plugin, so this issue shouldn’t apply. Moreover, our array-like authors.yml works perfectly fine with our existing layouts and includes, and has done so for many hundreds of builds prior to last week.
 
Besides, I think that issue would have broken a local build, and I will reiterate that calling `bundle exec jekyll build` using the github-pages gem version 155 returns a complete site without any warnings or errors: https://travis-ci.org/programminghistorian/jekyll/builds/265767247
 
Are there any specific configurations in the GitHub Pages environment for Jekyll that would return a generic guild error, but NOT a local build error using the github-pages gem?
 
-Matt
 
From: "Shawna Jean (GitHub Staff)" <support@github.com>
Date: Saturday, August 19, 2017 at 2:55 PM
To: Matthew Lincoln <MLincoln@getty.edu>
Subject: Re: gh-pages local build success; github.com build failure
 
Hey Matt,

Are there any updates on the cause of this generic page build? None of the listed problems at https://help.github.com/articles/generic-jekyll-build-failures/ seem to apply here.

Our devs were able to take a look and it turns out that your authors file is in the wrong format:

https://github.com/programminghistorian/jekyll/blob/gh-pages/_data/authors.yml

It's an array, not a hash:

https://jekyll.github.io/jekyll-seo-tag/advanced-usage/#author-information

Can you update your authors file to match the example above?

Let us know if you still run into build errors.

Best,
Shawna

**Comment by mdlincoln** [comment]
Hey Matthew,

Without another theme set in your config, our default theme will be used. The default theme was recently updated to use the jekyll-seo-tag plugin, which is why this error suddenly appeared.

It's possible that the github-pages gem doesn't currently fall back to the Primer theme when building locally. I'm not certain on that, so I've passed that question along to the team as well as asking for some more information about how you would be able to avoid using the default theme's config in this instance. As soon as we hear back, we'll follow up with you.

Best,
Shawna

**Comment by mdlincoln** [comment]
So I took a wild stab in the dark: given that a _secretly-loaded_ plugin was misusing our `authors.yml` file... I just renamed the file to `ph_authors.yml` (and also renamed all machine-
 and human-readable references to it across the site). This seems to have fixed our build, so now <https://programminghistorian.org/lessons/fetch-and-parse-data-with-openrefine> is live!

*cue 👼 choir*

I'm going to follow up with the GitHub support folks to let them know about this fix, but also suggest they fix their documentation and error reporting. Frankly, if there are going to be reserved names like this, it needs to be documented, and the error messages need to be informative! But I hope this current workaround is acceptable to everyone.
